src/routing_table/prefix.rs

```diff
@@ -73,6 +73,12 @@ impl<T: Clone + Copy + Default + Binary + Xorable> Prefix<T> {
         i >= self.bit_count() || i >= other.bit_count()
     }
 
+    /// Returns `true` if `other` is compatible but strictly shorter than `self`.
+    pub fn extends(&self, other: &Prefix<T>) -> bool {
```

Comment:
Ideally maybe the type should be called `BitSeq` or something, then I'd be happy with `is_prefix_of`.
But I think it's weird to say that a prefix is a prefix of a prefix…